File: Backend/retail_management_system/src/main/java/com/rms/model/Wholesaler.java
package com.rms.model;

import com.rms.model.Product;
import com.rms.model.User;
import jakarta.persistence.*;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Entity
@Table(name = "wholesalers")
@Data
@RequiredArgsConstructor
public class Wholesaler {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToMany(mappedBy = "wholesaler")
    private List<Product> products;

    private String businessName;
    private String address;
    private String gstNumber;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: Frontend/src/types/wholesaler.ts
export interface WholesalerProfile {
  id: number;
  username: string;
  email: string;
  phone: string;
  businessName: string;
  address: string;
  gstNumber: string;
  isActive: boolean;
}

export interface UpdateWholesalerDTO {
  username?: string;
  phone?: string;
  businessName?: string;
  address?: string;
  gstNumber?: string;
}